openghg/store/_flux.py

```diff
@@ -228,20 +212,23 @@ def read_file(
         if optional_metadata is None:
             optional_metadata = {}
 
-        # Make sure none of these are Nones
-        to_add = {k: v for k, v in optional_keywords.items() if v is not None}
-
-        # warn if `optional_metadata` overlaps with keyword arguments
-        overlap = [k for k in optional_metadata if k in to_add]
-        if overlap:
-            msg = (
-                f"Values for {', '.join(overlap)} in `optional_metadata` are "
-                "being overwritten by values passed as keyword arguments."
-            )
-            logger.warning(msg)
+        # TODO: REMOVE ONCE OPTIONAL KEYS CAN BE TAKEN FROM OTHER INPUTS THAN OPTIONAL_METADATA
+        optional_keywords: dict[Any, Any] = {
+            "database": database,
+            "database_version": database_version,
+            "model": model,
+        }
 
-        # update `optional_metadata` dict with any "optional" arguments passed to the parser
+        to_add = {key: value for key, value in optional_keywords.items() if value is not None}
         optional_metadata.update(to_add)
+        ######
+
+        # Check to ensure no required keys are being passed through optional_metadata dict
+        if optional_metadata is not None:
+            additional_metadata.update(optional_metadata)
+
+        # Mop up and add additional keys to metadata which weren't passed to the parser
+        flux_data = self.update_metadata(flux_data, additional_input_parameters, additional_metadata)
 
         lookup_keys = self.get_lookup_keys(optional_metadata=optional_metadata)
 
```

Comment:
Lines 235-237 can be removed

Reply:
I see that change could be applied on this PR but think those lines are removed in PR #1112 ? (and in original PR #1069 ). May be easiest to keep that change there?